src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/behandlingsresultat/BehandlingsresultatUtils.kt

```diff
@@ -0,0 +1,51 @@
+
+import no.nav.familie.ks.sak.common.exception.FunksjonellFeil
+import no.nav.familie.ks.sak.kjerne.behandling.domene.Behandling
+import no.nav.familie.ks.sak.kjerne.behandling.domene.BehandlingType
+import no.nav.familie.ks.sak.kjerne.behandling.domene.Behandlingsresultat
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.domene.Vilkårsvurdering
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.validerBarnasVilkår
+import no.nav.familie.ks.sak.kjerne.beregning.EndretUtbetalingAndelMedAndelerTilkjentYtelse
+import no.nav.familie.ks.sak.kjerne.beregning.EndretUtbetalingAndelValidator
+import no.nav.familie.ks.sak.kjerne.beregning.EndretUtbetalingAndelValidator.validerAtAlleOpprettedeEndringerErUtfylt
+import no.nav.familie.ks.sak.kjerne.beregning.TilkjentYtelseValidator
+import no.nav.familie.ks.sak.kjerne.beregning.domene.TilkjentYtelse
+import no.nav.familie.ks.sak.kjerne.personopplysninggrunnlag.domene.PersonopplysningGrunnlag
+
+object BehandlingsresultatUtils {
+
+    fun validerAtBehandlingsresultatKanUtføres(
+        vilkårsvurdering: Vilkårsvurdering,
+        personopplysningGrunnlag: PersonopplysningGrunnlag,
+        tilkjentYtelse: TilkjentYtelse,
+        endretUtbetalingMedAndeler: List<EndretUtbetalingAndelMedAndelerTilkjentYtelse>
+    ) {
+        val behandling = vilkårsvurdering.behandling
+        // valider vilkårsvurdering
+        if (behandling.type != BehandlingType.TEKNISK_ENDRING) {
+            validerBarnasVilkår(vilkårsvurdering, personopplysningGrunnlag.barna)
+        }
+        // valider TilkjentYtelse // TODO ikke komplett ennå, implementeres det med Sats
+        TilkjentYtelseValidator.validerAtTilkjentYtelseHarFornuftigePerioderOgBeløp(tilkjentYtelse, personopplysningGrunnlag)
+
+        // valider EndretUtbetalingAndel
+        validerAtAlleOpprettedeEndringerErUtfylt(endretUtbetalingMedAndeler.map { it.endretUtbetaling })
+        EndretUtbetalingAndelValidator.validerAtEndringerErTilknyttetAndelTilkjentYtelse(endretUtbetalingMedAndeler)
+    }
+
+    fun validerUtledetBehandlingsresultat(behandling: Behandling, behandlingsresultat: Behandlingsresultat) {
+        when {
+            behandling.type !in behandlingsresultat.gyldigeBehandlingstyper -> {
+                val feilmelding = "Behandlingsresultatet ${behandlingsresultat.displayName.lowercase()} " +
+                    "er ugyldig i kombinasjon med behandlingstype '${behandling.type.visningsnavn}'."
+                throw FunksjonellFeil(frontendFeilmelding = feilmelding, melding = feilmelding)
+            }
+            behandlingsresultat.erAvslått() && behandling.erKlage() -> {
+                val feilmelding = "Behandlingsårsak ${behandling.opprettetÅrsak.visningsnavn.lowercase()} " +
+                    "er ugyldig i kombinasjon med resultat '${behandlingsresultat.displayName.lowercase()}'."
+                throw FunksjonellFeil(frontendFeilmelding = feilmelding, melding = feilmelding)
```

Comment:
Disse to meldingene vises fram i frontend, men er behandlingsresultat noe SB velger selv?
Hvis ikke så kan det fort gjøre dem forvirret at disse meldingene dukker opp i frontend.
Hva skal de egentlig gjøre hvis dette oppstår? Kontakte team-baks?

Reply:
Disse valideringene finnes i ba-sak, her skrevet jeg det kun på en forbedret måte. SB kan ikke velge behandlingsresultat selv, men det er en sikkerhetsnett som sjekker om utledet behandlingsresultat samsvarer med behandlingstype og årsak, Hvis det feiler, bør SB kontakte team-baks.